h2/connection.py

```diff
@@ -1015,8 +1018,6 @@ def reset_stream(self, stream_id, error_code=0):
         stream = self._get_stream_by_id(stream_id)
         frames = stream.reset_stream(error_code)
         self._prepare_for_sending(frames)
-        self._reset_streams.add(stream_id)
-        del self.streams[stream_id]
```

Comment:
Before this PR, H2Connection need record that if it had sent RST_STREAM frame, so that H2Connection can handle receiving frames that just after we reset the stream.
But after this PR, this responsibility we will let H2StreamStateMachine and H2Stream take care for us, so it is no need to record reset_stream when we reset.